### PR TITLE
Rename files and structs

### PR DESCRIPTION
**Pull request by muneebahmed10**
*Description*:

Prefix all filenames with `core_mqtt_agent_`, and disambiguate structs with the `MQTTAgent` prefix.

### CODE REVIEW

File: test/cbmc/proofs/MQTTAgent_CommandLoop/Makefile [hunk 1]
```diff
@@ -48,6 +48,6 @@ PROOF_SOURCES += $(SRCDIR)/test/cbmc/stubs/agent_message_stubs.c
 PROOF_SOURCES += $(SRCDIR)/test/cbmc/stubs/agent_command_functions_stub.c
 PROOF_SOURCES += $(SRCDIR)/test/cbmc/stubs/core_mqtt_stubs.c
 
-PROJECT_SOURCES += $(SRCDIR)/source/mqtt_agent.c
```
Comment (gshvang) [inline test/cbmc/proofs/MQTTAgent_CommandLoop/Makefile:51]: Lines 37-39 needs to be updated from __CPROVER_file_local_mqtt_agent_c to __CPROVER_file_local_core_mqtt_agent_c

File: test/cbmc/proofs/MQTTAgent_ResumeSession/Makefile [hunk 2]
```diff
@@ -45,6 +45,6 @@ PROOF_SOURCES += $(SRCDIR)/test/cbmc/stubs/agent_command_pool_stubs.c
 PROOF_SOURCES += $(SRCDIR)/test/cbmc/stubs/agent_message_stubs.c
 PROOF_SOURCES += $(SRCDIR)/test/cbmc/stubs/core_mqtt_stubs.c
 
-PROJECT_SOURCES += $(SRCDIR)/source/mqtt_agent.c
```
Comment (gshvang) [inline test/cbmc/proofs/MQTTAgent_ResumeSession/Makefile:48]: Lines 34-36 needs to be updated from __CPROVER_file_local_mqtt_agent_c to __CPROVER_file_local_core_mqtt_agent_c

File: test/cbmc/proofs/MQTTAgent_CommandLoop/Makefile (hunk 1 above)
Comment (muneebahmed10) [reply test/cbmc/proofs/MQTTAgent_CommandLoop/Makefile:51]: Thanks, updated

File: source/include/core_mqtt_agent.h [hunk 3]
```diff
@@ -21,7 +21,7 @@
  */
 
 /**
- * @file mqtt_agent.h
+ * @file core_mqtt_agent.h
  * @brief Functions for running a coreMQTT client in a dedicated thread.
  */
 #ifndef MQTT_AGENT_H
```
Comment (gshvang) [inline source/include/core_mqtt_agent.h]: Should these names be changed also?

Comment (muneebahmed10) [reply source/include/core_mqtt_agent.h]: Yes, updated

**Review by gshvang — APPROVED** [review]

**Review by aggarg — APPROVED** [review]

**Pull request merged**